### PR TITLE
Multiple save file support

### PR DESCRIPTION
**Pull request by RedSparr0w**
Can have up to 9 different saves loaded at a time.

TODO:
- [x] Allow customization
- [x] Make look Better
- [x] Add Discord/Wiki/GitHub links to main screen
- [x] Limit to 9 save files
- [x] Show currently active challenges
- [x] Maybe some other stuff I am forgetting at the moment

_included a fix for downloading saves with characters outside of the Latin1 range in user categories_

Current Examples:
image
image
image

### CODE REVIEW

**Comment by Piotisgood** [comment]
Untitled
This is obviously not a finished piece, but I feel like having 3 different backgrounds would look better than just purple zigzags.
You dont have to use my one, but maybe different colors or shapes would look really good.

**Comment by RedSparr0w** [comment]
That is the plan, although I want to also make it so the user can decide their own background and stuff, but for now it's random:
image
image
image

**Comment by KrushGames** [comment]
Just going to assume, but would we be able to name these saves or not? like to give them labels we can identify them by, or would we need to keep track somewhere else?

**Comment by RedSparr0w** [comment]
Yeah, going to _try_ make it so you can change the:
- Name
- Trainer Icon
- Background
- Text Color
- Maybe add a favorite Pokémon or 2 or 4 or 6

File: src/components/saveSelector.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            <img style="width:100%" src="assets/images/pokeclickerlogo.png">
+        </div>
+        <div class="col-12 d-flex align-items-center justify-content-center my-3">
+            <a href="https://discord.gg/a6DFe4p">
+                <img alt="Join our Discord!" src="https://img.shields.io/discord/450412847017754644?color=7289DA&label=Discord&logo=discord&logoColor=7289DA&style=for-the-badge">
+            </a>
+            &nbsp;
+            <a href="https://github.com/pokeclicker/pokeclicker">
+                <img alt="Visit our GitHub!" src="https://img.shields.io/github/package-json/v/pokeclicker/pokeclicker/develop?label=GitHub&logo=github&style=for-the-badge">
+            </a>
```
Comment (HLXII) [inline src/components/saveSelector.html:13]: Is this pointing to the current version, or the develop version?

File: src/index.html [hunk 2]
```diff
@@ -84,6 +102,9 @@
 <!-- Get Keyitem modal-->
 @import "getKeyItem.html"
 
+<!-- Show keyitems modal-->
```
Comment (HLXII) [inline src/index.html:105]: Minor gripe on updating comments

**Review by HLXII — COMMENTED** [review]
Would be nice to see the active challenges on the trainer cards as well, maybe some additional badges/icons for each challenge? This might be something to be added later though.

File: src/components/saveSelector.html (hunk 1 above)
Comment (RedSparr0w) [reply src/components/saveSelector.html:13]: It's pointing to the dev version, live version would probably cause less confusion though I guess 😅

**Comment by RedSparr0w** [comment]
If anyone thinks any other ribbons would work better for the challenges, just download them from here and let me know:
https://msikma.github.io/pokesprite/overview/misc.html

(preferably gen 8 ribbons)

**Pull request merged**